rs-matter/src/secure_channel/case.rs

```diff
@@ -263,85 +283,50 @@ impl Case {
 
                 let encrypted_len = Case::get_sigma2_encryption(
                     fabric,
-                    exchange.matter.rand,
+                    exchange.matter().rand,
                     &our_random,
                     case_session,
                     signature,
                     encrypted_mut,
                 )?;
 
-                let encrypted = &encrypted[0..encrypted_len];
-
-                // Generate our Response Body
-                tx.reset();
-                tx.set_proto_id(PROTO_ID_SECURE_CHANNEL);
-                tx.set_proto_opcode(OpCode::CASESigma2 as u8);
-
-                let mut tw = TLVWriter::new(tx.get_writebuf()?);
-                tw.start_struct(TagType::Anonymous)?;
-                tw.str8(TagType::Context(1), &our_random)?;
-                tw.u16(TagType::Context(2), local_sessid)?;
-                tw.str8(TagType::Context(3), &case_session.our_pub_key)?;
-                tw.str16(TagType::Context(4), encrypted)?;
-                tw.end_container()?;
-
-                case_session.tt_hash.update(tx.as_mut_slice())?;
-
-                true
+                Some(encrypted_len)
             } else {
-                false
+                None
             }
         };
 
-        if fabric_found {
-            exchange.exchange(tx, rx).await
+        if let Some(encrypted_len) = encrypted_len {
+            let mut hash_updated = false;
+            let encrypted = &encrypted[0..encrypted_len];
+
+            exchange
+                .send_with(|_, wb| {
+                    let mut tw = TLVWriter::new(wb);
+                    tw.start_struct(TagType::Anonymous)?;
+                    tw.str8(TagType::Context(1), &our_random)?;
```

Comment:
Could we maybe create a struct that has derive ToTLV here to have things/types more readable?